backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DefaultDunningProducer.java

```diff
@@ -312,4 +343,10 @@ public void addAggregator(IDunningAggregator aggregator)
 		dunningAggregators.addAggregator(aggregator);
 	}
 
+	@Override
+	public Collection<DunningDocId> getCreatedDunningDocIds()
+	{
+		return isFinished ? createdDunningDocIds : Collections.emptySet();
```

Comment:
what's the reason of checking isFinished?

Reply:
Idea was to prevent getting an incomplete list of dunningDocs by calling this method before the aggregator has completed its work. Changed to `Check#assume()` to make the intention more obvious.

Reply:
don't see the change